### PR TITLE
Jump to generated func

### PR DESCRIPTION
**Pull request by rylev**
Worked on this with @yoshuawuyts.

We thought we ran into an issue with the `generate_function` assist where the code was not being generated in a certain situations. However, it wasn't actually a bug just a very confusing implementation where the cursor is not moved to the newly generated function. This happened when the return type was successfully inferred (and not unit). The function would be generated, but selection would not be changed.

This can be very confusing: If the function is generated somewhat far from where the assist is being invoked, the user never sees that the code was generated (nor are they given the chance to actually implement the function body). 

This PR makes it so that the cursor is _always_ moved to somewhere in the newly generated function. In addition, if we can infer unit as the type, then we do not still generate `-> ()` as the return type. Instead, we simply omit the return type.

This means the selection will move to either one of two places:
* A generated `-> ()` return type when we cannot successfully infer the return type.
* The `todo!()` body when we can successfully infer the return type.

### CODE REVIEW

**Comment by matklad** [comment]
bors d=@mahdi-frms

File: crates/ide_assists/src/handlers/generate_function.rs [hunk 1]
```diff
@@ -228,41 +239,16 @@ impl FunctionBuilder {
...
-        //
-        // If we are able to infer some return type, and that return type is not unit, we
-        // don't want to render the snippet. The assumption here is in this situation the
-        // return type is just as likely to be correct as any other part of the generated
-        // function.
-        //
-        // In the case where the return type is inferred as unit it is likely that the
-        // user does in fact intend for this generated function to return some non unit
-        // type, but that the current state of their code doesn't allow that return type
-        // to be accurately inferred.
```
Comment (mahdi-frms) [inline crates/ide_assists/src/handlers/generate_function.rs:242]: These should be rewritten for the `make_return_type` function.

**Review by mahdi-frms — COMMENTED** [review]
lgtm generally, but I think the documentation for the previous return type inference logic which is definitely not useful anymore, should be rewritten for the new `make_return_type` function.

**Comment by mahdi-frms** [comment]
bors r+

thanks!

**Comment by matklad** [comment]
Hm, not sure why d= didn't work

bors r=mahdi-frms

**Comment by mahdi-frms** [comment]
@matklad maybe you should'nt have written my id with a '@' symbol?

**Comment by lnicola** [comment]
Or it just missed the comment. It usually posts a comment along the lines of "Hey, mahdi-frms, use `bors r+` to merge this".

**Pull request merged**